portalnet/src/overlay_service.rs

```diff
@@ -873,78 +862,37 @@ where
                         tokio::spawn(async move {
-                            metrics.report_utp_active_inc(UtpDirectionLabel::Inbound);
-                            let mut stream = match utp
-                                .connect_with_cid(cid.clone(), *UTP_CONN_CFG)
-                                .await
-                            {
-                                Ok(stream) => stream,
-                                Err(err) => {
-                                    metrics.report_utp_outcome(
-                                        UtpDirectionLabel::Inbound,
-                                        UtpOutcomeLabel::FailedConnection,
-                                    );
-                                    debug!(
-                                        %err,
-                                        cid.send,
-                                        cid.recv,
-                                        peer = ?cid.peer.client(),
-                                        "Unable to establish uTP conn based on Content response",
-                                    );
+                            let trace = query_info.trace;
+                            let cid = utp_rs::cid::ConnectionId {
+                                recv: connection_id,
+                                send: connection_id.wrapping_add(1),
+                                peer: UtpEnr(source),
+                            };
+                            let data = match utp_controller.connect_inbound_stream(cid).await {
+                                Ok(data) => data,
+                                Err(e) => {
                                     if let Some(responder) = callback {
-                                        let _ = responder
-                                            .send(Err(OverlayRequestError::ContentNotFound {
-                                            message:
-                                                "Unable to locate content on the network: unable to establish utp conn"
-                                                    .to_string(),
-                                            utp: true,
-                                            trace: query_info.trace,
-                                        }));
-                                    };
+                                        let _ = responder.send(Err(
+                                            OverlayRequestError::ContentNotFound {
+                                                message: format!(
+                                                    "Unable to locate content on the network: {e}"
+                                                ),
+                                                utp: true,
+                                                trace,
+                                            },
+                                        ));
+                                    }
```

Comment:
`connect_inbound_stream` returns this error I would be a fan to pass trace through `connect_inbound_stream` just use the error returned directly instead of wrapping it twice

Reply:
We create `OverlayRequestError::ContentNotFound {}` in `connect_inbound_stream` and here I think we should remove the one here as doing it in both places is redundent.